backend/src/Designer/EventHandlers/ProcessTaskIdChanged/ProcessTaskIdChangedLayoutSetsHandler.cs

```diff
@@ -23,44 +27,128 @@ public ProcessTaskIdChangedLayoutSetsHandler(IAltinnGitRepositoryFactory altinnG
+                notification.EditingContext,
+                SyncErrorCodes.LayoutSetsTaskIdSyncError,
+                "App/ui/layout-sets.json",
+                async () =>
                 {
-                    return hasChanges;
-                }
+                    bool hasChanged = false;
+
+                    var layoutSets = await repository.GetLayoutSetsFile(cancellationToken);
+                    if (TryChangeLayoutSetTaskIds(layoutSets, notification.OldId, notification.NewId))
+                    {
+                        await repository.SaveLayoutSets(layoutSets);
+                        hasChanged = true;
+                    }
 
-                var layoutSets = await repository.GetLayoutSetsFile(cancellationToken);
-                if (TryChangeTaskIds(layoutSets, notification.OldId, notification.NewId))
+                    return hasChanged;
+                });
+
+            await ProcessLayouts(layoutSetName, notification, repository, cancellationToken);
+        }
+    }
+
+    private async Task ProcessLayouts(string layoutSetName, ProcessTaskIdChangedEvent notification,
+        AltinnAppGitRepository repository, CancellationToken cancellationToken)
+    {
+        string[] layoutNames;
+        try
+        {
+            layoutNames = repository.GetLayoutNames(layoutSetName);
+        }
+        catch (FileNotFoundException)
+        {
+            return;
+        }
+
+        foreach (string layoutName in layoutNames)
+        {
+            string layoutPath = $"App/ui/{layoutSetName}/{layoutName}.json";
+
+            await _fileSyncHandlerExecutor.ExecuteWithExceptionHandlingAndConditionalNotification(
+                notification.EditingContext,
+                SyncErrorCodes.LayoutTaskIdSyncError,
+                layoutPath,
+                async () =>
                 {
-                    await repository.SaveLayoutSets(layoutSets);
-                    hasChanges = true;
-                }
+                    bool hasChanged = false;
+                    var layout = await repository.GetLayout(layoutSetName, layoutName, cancellationToken);
+
+                    if (TryChangeLayoutTaskIds(layout, notification.OldId, notification.NewId))
+                    {
+                        await repository.SaveLayout(layoutSetName, layoutName, layout, cancellationToken);
+                        hasChanged = true;
+                    }
 
-                return hasChanges;
-            });
+                    return hasChanged;
+                });
+        }
     }
 
-    private static bool TryChangeTaskIds(LayoutSets layoutSets, string oldId, string newId)
+    private static bool TryChangeLayoutSetTaskIds(LayoutSets layoutSets, string oldId, string newId)
+    {
+        string originalLayoutSet = JsonSerializer.Serialize(layoutSets);
+        UpdateLayoutSetTaskIds(layoutSets, oldId, newId);
+        return !JsonSerializer.Serialize(layoutSets).Equals(originalLayoutSet);
```

Comment:
I suggest keeping the previous implementation where a boolean is returned from `UpdateLayoutSetTaskIds`, as it is likely more efficient than serializing and comparing the layout sets:

```suggestion
        string originalLayoutSet = JsonSerializer.Serialize(layoutSets);
        return UpdateLayoutSetTaskIds(layoutSets, oldId, newId);
```

Reply:
Agreed, and fixed.